samples/javascript/21.react-media-template/src/live-share-hooks/useSharedObjects.js

```diff
@@ -23,10 +23,13 @@ import { LiveShareHost } from "@microsoft/teams-js";
  * @returns Shared objects managed by the apps fluid container.
  */
 export function useSharedObjects() {
+    const startedRef = useRef(false);
     const [results, setResults] = useState();
     const [error, setError] = useState();
 
     useEffect(() => {
+        if (startedRef.current) return;
+        startedRef.current = true;
```

Comment:
As discussed, consider adding a statement in readme and/or in our live-share-react library ('s readme) that we are closely tracking new developments in React 18 with regard to `useEffect` and `Suspense` for data fetching so we will update when appropriate/widely supported.

Reply:
I added a comment here, I think that makes more sense than something in the project README, since this is a pretty isolated part of the project & this note doesn't impact the project at large. I will add a note to the README of the React library though, hopefully can spur some productive conversations with the community.